src/components/TodoItem.tsx

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todo: Todo;
+  isLoading?: boolean;
+  onDelete?: (todoId: number) => void;
+  onToggle?: (todoId: number, completed: boolean) => void;
+  onUpdate?: (todoId: number, title: string) => Promise<void>;
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  isLoading = false,
+  onDelete,
+  onToggle,
+  onUpdate,
+}) => {
+  const [isEditing, setIsEditing] = React.useState(false);
+  const [editTitle, setEditTitle] = React.useState(todo.title);
```

Comment:
Minor - but the style is inconcistent

```suggestion
  const [isEditing, setIsEditing] = useState(false);
  const [editTitle, setEditTitle] = useState(todo.title);
```